### PR TITLE
[incubator/jaeger] Made it possible to overwrite --reporter.grpc.host-port flag

### PR DESCRIPTION
**Pull request by qvistgaard**
#### What this PR does / why we need it:
It was not possible to change the --reporter.grpc.host-port to a custom hostname in case
the agent should send to a different service name.

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [X] [DCO](https://github.com/helm/charts/blob/master/CONTRIBUTING.md#sign-your-work) signed
- [X] Chart Version bumped
- [X] Title of the PR starts with chart name (e.g. `[stable/mychartname]`)

### CODE REVIEW

**Comment by naseemkullah** [comment]
Thanks for the PR. Can you confirm that the following is useless?  https://github.com/helm/charts/blob/8fae26093da93c588463bd86d5a87739afd37f55/incubator/jaeger/values.yaml#L112-L114 and if so also remove them, as they seem to be intended for this purpose.

File: incubator/jaeger/templates/agent-ds.yaml [hunk 1]
```diff
@@ -49,8 +49,10 @@ spec:
         image: {{ .Values.agent.image }}:{{ .Values.tag }}
         imagePullPolicy: {{ .Values.agent.pullPolicy }}
         env:
+        {{- if not (hasKey .Values.agent.cmdlineParams "reporter.grpc.host-port") }}
         - name: REPORTER_GRPC_HOST_PORT
           value: {{ include "jaeger.collector.name" . }}:{{ .Values.collector.service.grpcPort }}
```
Comment (naseemkullah) [inline incubator/jaeger/templates/agent-ds.yaml]: I forget, where does `"jaeger.collector.name" ` come from again?

Comment (naseemkullah) [reply incubator/jaeger/templates/agent-ds.yaml]: nevermind.

**Comment by naseemkullah** [comment]
Also even though you took the other PR into account and bumped up to `0.17.4`, still need to resolve the conflict.

**Comment by ssoerensen** [comment]
I Removed the entries from values.yaml. The daemon sets references a template function for generating the collector deployment name, which actually makes a lot of sense if the agent is deployed with the collector

**Comment by ssoerensen** [comment]
@naseemkullah I'm a bit unsure about how to handle the failing job, could you please advise?

**Review by naseemkullah — APPROVED** [review]

**Comment by naseemkullah** [comment]
Hi @ssoerensen, would you mind re creating this commit in this repo https://github.com/jaegertracing/helm-charts/tree/master/charts/jaeger ? We are in the process of moving, a deprecation notice will be added to this repo soon about this.

**Comment by davidkarlsen** [comment]
This chart has been deprecated / moved: https://github.com/helm/charts/commit/42892c2304f89d389ab7cd747d591d876fadb0e5

**Comment by ssoerensen** [comment]
sure, No problem

**Pull request closed without merging**